AI/VCAI/Goals/Build.cpp

```diff
@@ -327,7 +327,7 @@ Tasks::TaskList Build::getTasks()
 		}
 
 		double daysToGetResource = requiredResources[resType] / (dailyIncome[resType] + 0.001); // avoid zero divide
-		double priority = resType == std::max(1.0, daysToGetResource / 10);
+		double priority = resType == std::min(1.0, daysToGetResource / 10);
 
```

Comment:
This is incorrect.

If daysToGetResource is 0, we should get max priority (now 1) and realize the task immediately. If daysToGetResource is higher, the task should get lower priority. Now we get highest priority it it takes 10 or more days to collect resources.

Reply:
Exactly. As more days we need to get resource as more we are blocked by it and can't develop town further and need to do something to reduce this blocking. daysToGetResource are calculated as a ratio required/income. 0 means that we do not need it at all.

Reply:
But pro players are NOT blocked by town development. Developing towns over many days is suboptimal strategy to begin with.

By making AI fixed on the goals which are hardest to achieve, you will never make it progress.